packages/server/src/fhir/repo.ts

```diff
@@ -2570,7 +2570,7 @@ export class Repository extends FhirRepository<PoolClient> implements Disposable
     return this.context;
   }
 
-  [Symbol.dispose](): void {
+  [Symbol.dispose](removeConnection?: boolean): void {
```

Comment:
We should note that normally your `[Symbol.dispose]()` should not contain args since the symbol is intended to be used by automatic resource cleanup via [`using` keyword](https://simonkoeck.medium.com/typescript-5-2s-using-keyword-managing-resources-with-ease-e0a31efa3eec)

Eventually we should clean this up

Reply:
Yeah...I didn't love adding that, but I wasn't sure of a better way to achieve it for now. Very open to suggestions!

Reply:
We could add a new method on `Repository` to set the state to remove the connection on disposal I suppose, but I'm not sure how much better that is.